### ISSUE TITLE
Java webapp deployment fails on Azure Cloud Shell but works from Azure CLI

### ISSUE THREAD

**Issue by jalanubha** [Microsoft/azure-maven-plugins#358]
### Plugin name and version
name: azure-webapp-maven-plugin
version: 1.3.0

### Plugin configuration in your `pom.xml`
`

         <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>2.5.1</version>
                <inherited>true</inherited>
                <configuration>
                    <source>1.7</source>
                    <target>1.7</target>
                </configuration>
            </plugin>
            <plugin>
                <groupId>com.microsoft.azure</groupId>
                <artifactId>azure-webapp-maven-plugin</artifactId>
                
                <version>1.3.0</version>
                <configuration>
                    <resourceGroup>myrg</resourceGroup>
                    <appName>mywebapp</appName>
                </configuration>
            </plugin>

`

### Expected behavior
Web app should deploy successfully

### Actual behavior
Deployment fails with below error:

[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building simple-service-webapp 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- azure-webapp-maven-plugin:1.3.0:deploy (default-cli) @ simple-service-webapp ---
Cannot read files /home/bhanu/.azure/azureProfile.json and /home/bhanu/.azure/accessTokens.json. Are you logged in Azure CLI?
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 9.400 s
[INFO] Finished at: 2018-08-21T03:21:40+00:00
[INFO] Final Memory: 28M/206M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.microsoft.azure:azure-webapp-maven-plugin:1.3.0:deploy (default-cli) on project simple-service-webapp: Failed to authenticate with Azure. Please check your configuration. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

### Steps to reproduce the problem
Create a simple java web app using cloud shell and deploy using the Authentication Method 3 documented [here](https://github.com/Microsoft/azure-maven-plugins/blob/develop/docs/common-configuration.md)

Same steps when followed using Azure CLI on the local machine it deploys the app successfully.
Is this a known issue with the Cloud Shell?

**Comment by SummerSun** [comment]
@jalanubha Thank you for reporting the issue to us.

Could you verify that these two files do exist?
```/home/bhanu/.azure/azureProfile.json and /home/bhanu/.azure/accessTokens.json```

From my investigation, accessTokens.json is absent in cloud shell even after running ```az login```  manually. And that is why the authentication with azure-cli fails.

Issue https://github.com/Azure/azure-cli/issues/7107 was opened in azure-cli repository to track. And I would suggest you upload the local accessTokens.json file to Azure Cloud Shell to /home/bhanu/.azure/ unblock your work.

**Comment by jalanubha** [comment]
Thanks for checking @SummerSun 
I can see azureProfile.json but not accessTokens.json

**Comment by jalanubha** [comment]
It works after I upload accessTokens.json file to Azure Cloud Shell. Thanks for the workaround.

**Comment by JasonFreeberg** [comment]
Copy of #382

**Comment by jdneo** [comment]
Fixed in #558

**Comment by nodermatt** [comment]
@jalanhubha how did you create the accessTokens.json? I am having the same issue.

**Comment by nodermatt** [comment]
I found it in /tmp/ and copied the file to my .azure folder. Now deployment worked. Is that the intended procedure?

**Comment by Flanker32** [comment]
@nodermatt These files are created by Azure CLI, but accessTokens.json is missed in Azure cloud shell, 
this has partly been fixed in 1.5.3, now maven plugin could auth but will use the first subscription instead of the default one, this will be fixed in next release, as a workaround, you may set the subscription id in configuration, see #628.

**Comment by PhilStollery** [comment]
Can anyone help - getting this error in an Azure Cloud Shell, but can't find any documentation on how to setup MSI - should it be there automatically?

**Comment by Flanker32** [comment]
> Can anyone help - getting this error in an Azure Cloud Shell, but can't find any documentation on how to setup MSI - should it be there automatically?

Which version of maven plugin did you use? The latest version is 1.7.0, which should work in Cloud Shell.

**Comment by Chika20** [comment]
hello all,
I am facing same issue while deploying from CICD pipeline from Gitlab.
[ERROR] com.microsoft.azure.common.exceptions.AzureExecutionException: Server not found in settings.xml. ServerId=azure-auth

I have added new Service Principle details in settings.xml & provided all information.
Deployment working fine from cmd but not from script pipeline.

**Comment by Flanker32** [comment]
@Chika20 Thanks for your report, you need to update the credentials in settings.xml of pipeline, please refer [here](https://github.com/microsoft/azure-maven-plugins/wiki/Authentication#service-principles-in-settingsxml) for details. Or you may try set credential in pom configuration and set the values with environment variables.

**Comment by vnextcoder** [comment]
This Issue also happens if you are having AZ CLI installed within WSL 2 and running MVNW from WSL2 only.

For troubleshooting this on WSL , I tried all steps necessary to clear the Subscriptions, Set Default Subscription from Command Line and also set <subscriptionId></subscriptionId> in the pom.xml as well. 

thing is the Azure.Subscriptions.list() just returns nothing. it could be a bug within the Azure Management JAR itself, but I am not sure as of now. 

Moreover, I have also tried using Maven Azure Plugin 1.7, 1.9.1 and 1.10.0 as well. All failed on  WSL2

Once I switched to the Windows 10 native, it worked fine.
